src/script/util/KeyboardUtil.js

```diff
@@ -142,3 +142,19 @@ z.util.KeyboardUtil = (() => {
     isRemovalAction: _isRemovalAction,
   };
 })();
+
+const escKeyListeners = {};
+
+document.addEventListener('keydown', event => {
+  if (event.key === 'Escape') {
+    Object.values(escKeyListeners).forEach(handler => handler(event));
+  }
+});
+
+export const listenToEscKey = (identifier, handler) => {
+  escKeyListeners[identifier] = handler;
+};
+
+export const unlistenToEscKey = identifier => {
+  delete escKeyListeners[identifier];
+};
```

Comment:
I am going to tackle the rest in a separate PR

Reply:
Can you add tests for those new functions.
Should be quite simple